src/include/sof/coherent.h

```diff
@@ -117,18 +134,120 @@ static inline struct coherent *coherent_release(struct coherent *c, const size_t
 	return cache_to_uncache(c);
 }
 
-#define coherent_init(object, member)						\
-	do {									\
-		/* assert if someone passes a cache/local address in here. */	\
-		ADDR_IS_COHERENT(object);					\
-		/* TODO static assert if we are not cache aligned */		\
-		k_spinlock_init(&object->member.lock);				\
-		object->member.shared = false;					\
-		object->member.core = cpu_get_id();				\
-		list_init(&object->member.list);				\
-		/* inv local data to coherent object */				\
-		dcache_invalidate_region(uncache_to_cache(object), sizeof(*object)); \
-	} while (0)
+static inline void __coherent_init(struct coherent *c, const size_t size)
```

Comment:
Any reason why we are making this change to an inline ?

Reply:
to stop dragging those backslashes around at least... Not to mention reducing indentation depth, strict parameter type-checking, eventual move to a .c file, as this .h has already some too much code in it IMHO